src/java.base/share/classes/java/util/concurrent/locks/AbstractQueuedLongSynchronizer.java

```diff
@@ -369,14 +369,19 @@ final int acquire(Node node, long arg, boolean shared,
             } else if (node.status == 0) {
                 node.status = WAITING;          // enable signal and recheck
             } else {
-                long nanos;
                 spins = postSpins = (byte)((postSpins << 1) | 1);
-                if (!timed)
-                    LockSupport.park(this);
-                else if ((nanos = time - System.nanoTime()) > 0L)
-                    LockSupport.parkNanos(this, nanos);
-                else
-                    break;
+                try {
+                    long nanos;
+                    if (!timed)
+                        LockSupport.park(this);
+                    else if ((nanos = time - System.nanoTime()) > 0L)
+                        LockSupport.parkNanos(this, nanos);
+                    else
+                        break;
+                } catch (Error | RuntimeException ex) {
```

Comment:
@DougLea @AlanBateman Changed to Error | RuntimeException here and for AQLS

Reply:
lgtm!

Reply:
`catch (Throwable ex)` would be consistent with the similar block at line 331. Though I'm unclear how that compiles without the method declaring `throws Throwable` ??

Reply:
> Though I'm unclear how that compiles without the method declaring `throws Throwable` ??

It wouldn't need that because of precise rethrow. In any case, having Error and RuntimeException are okay.